### DISCUSSION TITLE
lock_settings from /room/getJoinToken are not being applied

### DISCUSSION THREAD

**Discussion by tecbr** [mynaparrot/plugNmeet-server#281] [General]
lock_settings are not being applied to the user when he joins the room via link generated by /room/getJoinToken

**Comment by jibon57** [comment]
Please use discussion section first before opening issue ticket. Lock setting will be applicable only for non admin users. For admin no lock will be applicable.

**Comment by tecbr** [comment]
I tested for non-admin users and the settings are not being applied.

**Reply by jibon57** [reply]
If you change lock setting after join the room then this new lock settings won't apply for the user because join token have different lock parameters. One thing you can do is set default lock settings during create room.

**Reply by tecbr** [reply]
I am not changing lock setting after join the room. I'm generating a token with lock_settings field exactly as defined here [https://www.plugnmeet.org/docs/api/join_room](https://www.plugnmeet.org/docs/api/join_room)
But these lock_settings have no effect.
Default lock settings during create room are working fine.

**Reply by jibon57** [reply]
Thanks for reporting. Will fix it.

**Reply by jibon57** [reply]
issue ticker: https://github.com/mynaparrot/plugNmeet-server/issues/282

**Reply by jibon57** [reply]
@tecbr will you please check from demo server? I've pushed one fix there.

**Reply by tecbr** [reply]
Now the lock setting is working except lock_private_chat because it was possible to send private message with both lock_private_chat = true on the default lock setting during create room and lock via token.

But I think lock_private_chat is not completely implemented because it does not appear in the fields listed in the documentation, I saw it only in the example.

Thanks for the fix.

**Reply by jibon57** [reply]
User can always send private messages to admin/moderator no matter what was setting & vise versa. Locking private chat will disable to send messages each other.

**Reply by tecbr** [reply]
OK. This makes sense.

After testing lock_private_chat between non admin users:
lock_private_chat in the creation of the room: working
lock_private_chat via token: not working.

**Reply by jibon57** [reply]
@tecbr thanks for pointing that. I've pushed one fix here: https://github.com/mynaparrot/plugNmeet-client/commit/27009dceb82114041af7c8967bacdc63bbcde84f
can you try from demo server again?

**Reply by tecbr** [reply]
lock_private_chat via token: working.

Thanks.
